src/pstl_offload.cpp

```diff
@@ -0,0 +1,137 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <new>
+#include <cassert>
+#include <cstdint>
+#include <sycl/sycl.hpp>
+
+#include <pstl_offload/internal/usm_memory_replacement_common.h>
+
+#if __linux__
+
+#define _PSTL_OFFLOAD_EXPORT __attribute__((visibility("default")))
+
+#include <dlfcn.h>
+#include <string.h>
+#include <unistd.h>
+
+namespace __pstl_offload {
+
+auto __get_original_free() {
```

Comment:
Why is it still not `static`? The same is true for the functions below. Did we change the intent? Did we face some issues?

Reply:
What is the value of getting these functions in `.cpp` file `static`? Symbol hiding would be achieved by hidden visibility attribute that would be used by default for each function not explicitly having `visibility(default)` that is guaranteed by the build flag that is set on the CMake side.

Reply:
Fixed